packages/dashboard/src/customization/widgetPropertiesGeneratorMap.ts

```diff
@@ -11,5 +12,6 @@ export const WidgetPropertiesGeneratorMap: {
     //eslint-disable-next-line
     properties?: () => Record<any, any>;
     initialSize?: Pick<AnyWidget, 'height' | 'width'>;
+    onUpdateWidget?: (action: DashboardAction) => DashboardAction;
```

Comment:
What if we just had a function which takes a widget type and returns a widget type. I don't think a user of the api should have to care about redux

Reply:
This was my idea with the useWidgetActions hook. The only thing it does right now is give you a function to update widget, you just need to call it with your updated widget and it handles dispatching the correct store action.

Reply:
updated